dht.go

```diff
@@ -288,6 +276,25 @@ func makeRoutingTable(dht *IpfsDHT, cfg config) (*kb.RoutingTable, error) {
 	return rt, err
 }
 
+// TODO This is hacky AND SHOULD be removed once https://github.com/libp2p/go-libp2p-core/pull/117
```

Comment:
@Stebalien Please can I get some eyeballs on this hacky function ?

Reply:
Thinking through this... This is horrible, it's hacky, but it _works_ and it definitely won't clobber _longer_ lifetimes.

However, instead of repeatedly re-adding the same addresses, I'd switch this to `dht.peerstore.UpdateAddrs(p, peerstore.RecentlyConnectedAddrTTL, peerstore.RecentlyConnectedAddrTTL)`. This will repeatedly extend the timeouts of all addresses in the "recently connected" state, but it won't extend the timeouts of other address (e.g., discovered addresses).

Reply:
Note: the _correct_ way to do this right now is to introduce a new custom TTL:

```
const (
	// PermanentAddrTTL is the ttl for a "permanent address" (e.g. bootstrap nodes).
	PermanentAddrTTL = math.MaxInt64 - iota

	// This range is reserved for marker TTLs
	// Higher-TTLs always outrank lower TTLs. Lower TTLs will never clobber
	// a higher TTL.
	DHTRoutingTableTTL
	OtherMarkerTTL

	// ConnectedAddrTTL is the ttl used for the addresses of a peer to whom
	// we're connected directly. This is basically permanent, as we will
	// clear them + re-add under a TempAddrTTL after disconnecting.
	ConnectedAddrTTL = math.MaxInt32
)
```

However, given that we're going to get address pinning as soon as this release is out, I'm not sure if this change is worth it.

Reply:
Actually, even that's wrong. Basically, there's no safe/sane way to do this right now other than to just repeatedly say "please keep these addresses".

Reply:
@Stebalien  

>  I'd switch this to dht.peerstore.UpdateAddrs(p, peerstore.RecentlyConnectedAddrTTL, peerstore.RecentlyConnectedAddrTTL)

That's a good idea. I've made the change and changed the documentation of this func to really show how bad this is.